src/shared/codelens/goCodeLensProvider.ts

```diff
@@ -0,0 +1,195 @@
+        const funcWithBody: string = document.getText(symbol.range)
+
+        const bodyStart: number = funcWithBody.search(/\{/)
+        const sigStart: number = funcWithBody.search(/\(/)
+
+        if (bodyStart === -1 || sigStart === -1) {
+            return false
+        }
+
+        const funcSig: string = stripNewLinesAndComments(funcWithBody.substring(sigStart, bodyStart))
+
+        const funcSigParts: string[] = funcSig.split(/\)/)
+        const argTypes: string[] = parseTypes(funcSigParts[0])
+        const retTypes: string[] = funcSigParts.length > 1 ? parseTypes(funcSigParts[1]) : []
+
+        if (argTypes.length > 2 || retTypes.length > 2) {
+            return false
+        }
+
+        // TODO: actually check the types to make sure they are valid
+
+        return true
+    }
+
+    return false
+}
+
+/**
+ * Go allows function signatures to be multi-line, so we should parse these into something more usable.
+ *
+ * @param text String to parse
+ *
+ * @returns Final output without any new lines or comments
+ */
+function stripNewLinesAndComments(text: string): string {
+    const blockCommentRegExp = /\/\*[.*?]\*\//
+    let result: string = ''
+
+    text.split(/\r|\n/).map(s => {
+        const commentStart: number = s.search(/\/\//)
+        s = s.replace(blockCommentRegExp, '')
+        result += commentStart === -1 ? s : s.substring(0, commentStart)
+    })
+
+    return result
+}
+
+/**
+ * Finds all types of the parameter list, stripping out names if they exist.
+ * Parenthesis are stripped from the list if they exist.
+ *
+ * @param params List of parameters delimeted by commas
+ *
+ * @returns A list of types mapping 1:1 with the input
+ */
+function parseTypes(params: string): string[] {
+    params = params.replace(/\(|\)/, '')
+    const types: string[] = []
+
+    let lastType: number = 0
+    const paramParts = params.split(',')
+
+    // Names of parameters must either be all present or all absent: https://golang.org/ref/spec#Function_types
+    for (let i = 0; i < paramParts.length; i++) {
+        const parts: string[] = paramParts[i].trim().split(/\s+/)
+        const type: string = parts.length > 1 ? parts[1] : parts[0]
+
+        // Go allows types to be assigned to multiple parameters, e.g. (x, y, z int) === (x int, y int, z int)
+        if (parts.length > 1) {
+            for (let j = lastType; j < i; j++) {
+                types[j] = type
+            }
+            lastType = i + 1
+        }
+
+        types.push(type)
+    }
+
+    return types
+}
+
+/**
+ * Checks if the Go extension exists and is active.
+ */
+async function checkForGoExtension(): Promise<boolean> {
+    const extension = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.go)
+
+    if (extension) {
+        if (extension.isActive) {
+            return true
+        }
+
+        getLogger().info('Go CodeLens provider is activating the Go extension...')
```

Comment:
Probably should be DEBUG level. Also let's use parallel form with the suggestions in https://github.com/aws/aws-toolkit-vscode/pull/1611/files#r605742852 .

```
getLogger().debug('Activating extension: %s', VSCODE_EXTENSION_ID.go )
```